rclcpp/include/rclcpp/function_traits.hpp

```diff
@@ -92,6 +92,18 @@ struct function_traits<
   : function_traits<ReturnTypeT(Args ...)>
 {};
 
+// std::bind for free functions
+template<typename ReturnTypeT, typename ... Args, typename ... FArgs>
+#if defined _LIBCPP_VERSION  // libc++ (Clang)
```

Comment:
Yeah, the C++11 spec doesn't define the return value of a `std::bind` call:

http://en.cppreference.com/w/cpp/utility/functional/bind

I agree with you, but I couldn't come up with a different approach. I tried using `decltype` and `declval` so that it was compiler-independent, but couldn't get it to work. This is the least intrusive way I can think of supporting `std::bind` across compilers.